### PR TITLE
#1604: Update CONTRIBUTING.md

### LINKED ISSUE #1604: Update CONTRIBUTING.md
## Background
`CONTRIBUTING.md` ought to be refreshed.

## Feature
Update _Feature/Bug_ chapter. Add _Documentation_ chapter.

### PR DESCRIPTION
**Pull request by benedeki**
Closes #1604

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -12,22 +12,36 @@
...
+* After finishing everything, push to your forked repo and open a Pull Request to our `develop` branch:
   * Pull Request titles should start with the Github Issue number:
     * **"42 Life, the universe and everything"**
   * Ensure the Pull Request description clearly describes the solution.
+  * Connect the PR to the _Issue_
+
+## **Do you want to improve the project's documentation?**
+
+* The process is similar, just a tad bit simpler, than the feature or bugfix implementation
+* For documentation changes an _Issue_ doesn't neet to exist. For bigger changes we still encourage to have one.
```
Comment (Zejnilovic) [inline CONTRIBUTING.md:36]:
```suggestion
* For documentation changes an _Issue_ doesn't need to exist. For bigger changes we still encourage to have one.
```

Comment (Zejnilovic) [inline CONTRIBUTING.md:36]: these are more like full sentences, not points

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -12,22 +12,36 @@
...
+* For the repository.
+  * You want to base your changes on the `gh-pages` branch.
+  * Name the branch with **ghp/** prefix, if a connected issue exists add its number after the **ghp/** prefix  
+    * **ghp/dont-panic**
+    * **ghp/42-dont-panic**  
+* After finishing, push to your forked repo and open a Pull Request to our `gh-pages` branch.
+  * Describe in the Pull Request description what the changes are about, what was their motivation.
```
Comment (Zejnilovic) [inline CONTRIBUTING.md:43]:
```suggestion
  * In the Pull Request describe what the changes are about, what was their motivation.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (benedeki) [reply CONTRIBUTING.md]: And are they mutually exclusive? 😉

Comment (Zejnilovic) [reply CONTRIBUTING.md]: Not exactly. You can have full sentences in your bullets, but expressing a paragraph of text in bullets is weird. You expect steps, not description. Or at least you expect steps and description in bullets to be differentiated

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -12,22 +12,36 @@
...
+* The process is similar, just a tad bit simpler, than the feature or bugfix implementation
+* For documentation changes an _Issue_ doesn't need to exist. For bigger changes we still encourage to have one.
+* For the repository.
+  * You want to base your changes on the `gh-pages` branch.
+  * Name the branch with **ghp/** prefix, if a connected issue exists add its number after the **ghp/** prefix  
+    * **ghp/dont-panic**
+    * **ghp/42-dont-panic**  
+* After finishing, push to your forked repo and open a Pull Request to our `gh-pages` branch.
+  * In the Pull Request describe what the changes are about, what was their motivation.
+  * Connect the PR to the _Issue_ if it exists
```
Comment (Zejnilovic) [inline CONTRIBUTING.md:44]:
```suggestion
The process is similar, just a tad bit simpler, than the feature or bugfix implementation. For documentation changes, an _Issue_ doesn't need to exist. For bigger changes, we still encourage to have one.

To implement documentation changes:
  * Fork the repository
  * Base your changes on the `gh-pages` branch.
      * Name the branch with **ghp/** prefix, if a connected issue exists add its number after the **ghp/** prefix  
        * **ghp/dont-panic**
        * **ghp/42-dont-panic**  
* After finishing, push to your forked repo and open a Pull Request to our `gh-pages` branch.
  * In the Pull Request describe what the changes are about, what was their motivation.
  * Connect the PR to the _Issue_ if it exists
```

**Review by dk1844 — APPROVED** [review]

**Review by Zejnilovic — APPROVED** [review]

**Pull request merged**